app/models/course/assessment/answer/programming.rb

```diff
@@ -43,4 +43,17 @@ def attempting_times_left
   def grade_inline?
     false
   end
+
+  def downloadable?
```

Comment:
> Would it make sense for this to be a class method instead?
Since the answers are instantiated, I think instance method is ok and easier for client code to use ( `answer.class.downloadble?` vs `answer.downloadable?` ). 

unless there's a way to not instantiate the answer ? that will improve some performance 😄